### PR TITLE
Fix release workflow

### PR DESCRIPTION
**Pull request by gimppa**
Revert changes to release.yml done in #36 to fix the windows noconsole bug.
Re-implement #38 without using mock-like approach and only targeting windows systems.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -78,12 +78,12 @@ jobs:
...
           echo '${{ secrets.DOCKER_PASSWORD }}' | docker login -u '${{ secrets.DOCKER_USERNAME }}' --password-stdin
           docker pull cscfi/pyinstaller
-          docker run --rm -v "${PWD}:/builder" cscfi/pyinstaller --onefile sdagui.py --name ${{ matrix.asset_name }}
+          docker run --rm -v "${PWD}:/builder" cscfi/pyinstaller --noconsole --onefile sdagui.py --name ${{ matrix.asset_name }}
           sudo chown -R $USER:$GROUP dist/
       - name: Build GUI artifact
         if: matrix.os != 'ubuntu-latest'
         run: |
-          pyinstaller --onefile sdagui.py --name ${{ matrix.asset_name }}
+          pyinstaller --noconsole --onefile sdagui.py --name ${{ matrix.asset_name }}
```
Comment (blankdots) [inline .github/workflows/release.yml:86]: i think here we need to keep --noconsole because of windows bug ? or I am getting it wrong ?

Comment (teemukataja) [reply .github/workflows/release.yml:86]: it was initially removed as a compromise, but fixed here https://github.com/CSCfi/sda-uploader/pull/38 so we can hide the cmd again for windows

**Review by teemukataja — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (blankdots) [reply .github/workflows/release.yml:86]: using mock to do that, not really ideal, i like this solution explored: https://stackoverflow.com/a/25061573

**Review by blankdots — CHANGES_REQUESTED** [review]
i prefer a different solution explored: https://stackoverflow.com/a/25061573

and maybe only on windows:

```
import platform
>>> platform.system()
'Windows'
```

**Comment by gimppa** [comment]
I re-implemented #38 without using mock and only targeting windows OS. Since the bug really only concerns the case where pyinstaller is run using the --noconsole flag which results in sys.stdout (and sys.stdout.write) not being defined, I don't think we need to redirect sys.stdout to devnull. 
Also, maybe I'm wrong here but it seems to me it would require extra work to get contextmanager to work with the existing code, and the windows OS condition.

**Comment by blankdots** [comment]
well if context managers are not a fit:
```
f = open(os.devnull, 'w') 
sys.stdout = f
```

what do you think?

Though your solution works as well, though looks a bit similar to what a mock is 🙂

**Comment by gimppa** [comment]
Re-implemented this without using the mock-like approach.

**Comment by blankdots** [comment]
still some style things to fix, other than that lgtm as code goes, hopefully someone manged to test in windows

**Comment by gimppa** [comment]
I'm wondering why the python style checks don't pass? Flake8 runs just fine locally but in the github automation the process exits with code 1 without any sort of indication what's wrong.

**Review by blankdots — APPROVED** [review]

**Pull request merged**